sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -0,0 +1,246 @@
+          should be computed.
+        Ignored in binary classification or classical regression settings.
+
+    pdp_lim : dict
+        Global min and max average predictions. `pdp_lim[1]` is the global min
+        and max for single partial dependence curves. `pdp_lim[2]` is the
+        global min and max for two-way partial dependence curves.
+
+    deciles : dict
+        Deciles for feature indicies in ``features``.
+
+    Attributes
+    ----------
+    bounding_ax_ : matplotlib Axes or None
+        If `ax` is an axes or None, the `bounding_ax_` is the axes where the
+        grid of partial dependence plots are drawn. If `ax` is a list of axe
+        or a numpy array of axes, `bounding_ax_` is None.
+
+    axes_ : ndarray of matplotlib Axes
+        If `ax` is an axes or None, `axes_[i, j]` is the axes on the i-th row
+        and j-th column. If `ax` is a list of axes, `axes_[i]` is the i-th item
+        in `ax`. Elements that are None corresponds to a nonexisting axes in
+        that position.
+
+    lines_ : ndarray of matplotlib Artists
+        If `ax` is an axes or None, `line_[i, j]` is the partial dependence
+        curve on the i-th row and j-th column. If `ax` is a list of axes,
+        `lines_[i]` is the partial dependence curve corresponding to the i-th
+        item in `ax`. Elements that are None corresponds to a nonexisting axes
+        or an axes that does not include a line plot.
+
+    contours_ : ndarray of matplotlib Artists
+        If `ax` is an axes or None, `contours_[i, j]` is the partial dependence
+        plot on the i-th row and j-th column. If `ax` is a list of axes,
+        `contours_[i]` is the partial dependence plot corresponding to the i-th
+        item in `ax`. Elements that are None corresponds to a nonexisting axes
+        or an axes that does not include a contour plot.
+
+    figure_ : matplotlib Figure
+        Figure containing partial dependence plots.
+
+    """
+    def __init__(self, pd_results, features, feature_names, target_idx,
+                 pdp_lim, deciles):
+        self.pd_results = pd_results
+        self.features = features
+        self.feature_names = feature_names
+        self.target_idx = target_idx
+        self.pdp_lim = pdp_lim
+        self.deciles = deciles
+
+    def plot(self, ax=None, n_cols=3, line_kw=None, contour_kw=None, fig=None):
+        """Plot partial dependence plots.
+
+        Parameters
+        ----------
+        ax : Matplotlib axes or array-like of Matplotlib axes, default=None
+            Axes to plot the partial dependence curves.
+            - If a single axes is given, it is treated as a bounding axes and
+                a grid of partial depedendence plots will be drawn within these
+                bounds.
+            - If a list of axes or a ndarray of axes are passed in, the partial
+                dependence plots will be drawn directly into these axes.
```

Comment:
maybe indicate that the translation from a 1d list to a 2d grid is done based on the `n_cols` parameter?

Reply:
I should be more clear here. If an array-like of axes is passed in, `n_cols` does nothing.